src/display/editor/annotation_editor_layer.js

```diff
@@ -91,13 +94,45 @@ class AnnotationEditorLayer {
     this.#uiManager.addLayer(this);
   }
 
+  /**
+   * The mode has changed: it must be updated.
+   * @param {number} mode
+   */
+  updateMode(mode) {
+    if (mode === AnnotationEditorType.INK) {
+      // We want to have the ink editor covering all the page
+      // and no click to create it: it must be here when we start to draw.
+      this.div.addEventListener("mouseover", this.#boundMouseover);
+      this.div.removeEventListener("click", this.#boundClick);
+    } else {
+      this.div.removeEventListener("mouseover", this.#boundMouseover);
+    }
+  }
+
+  /**
+   * mouseover callback.
+   * @param {MouseEvent} event
+   */
+  mouseover(event) {
+    if (event.target === this.div && event.buttons === 0) {
+      // The div is the target so there is no ink editor, hence we can
+      // create a new one.
+      // event.buttons === 0 is here to avoid adding a new ink editor
+      // when we drop and editor.
+      const editor = this.#createAndAddANewEditor(event);
+      editor.setInBackground();
+    }
+  }
+
   /**
    * Add some commands into the CommandManager (undo/redo stuff).
    * @param {function} cmd
    * @param {function} undo
+   * @param {boolean} mustExec - if true the command is executed after having
+   * been added.
    */
-  addCommands(cmd, undo) {
-    this.#uiManager.addCommands(cmd, undo);
+  addCommands(cmd, undo, mustExec = true) {
```

Comment:
Similar to another recent PR, can we please have the default value be `false` instead?
(Since having `undefined` => `true` feels quite surprising.)